CMakeLists.txt

```diff
@@ -46,59 +66,220 @@ generate_messages(DEPENDENCIES std_msgs sensor_msgs)
 catkin_package(
   INCLUDE_DIRS include
   LIBRARIES ${PROJECT_NAME}
-  CATKIN_DEPENDS roscpp image_transport sensor_msgs cv_bridge geometry_msgs tf message_filters std_msgs std_srvs 
-                 visualization_msgs dynamic_reconfigure pcl_ros message_runtime
-  DEPENDS SuiteSparse CVD OpenGL GVars3)
-
-add_executable(mcptam src/Main.cc src/GLWindow2.cc src/GLWindowMenu.cc src/VideoSourceMulti.cc src/CameraGroupSubscriber.cc 
-                      src/SystemBase.cc src/SystemFrontendBase.cc src/System.cc src/TaylorCamera.cc src/KeyFrame.cc src/MapPoint.cc src/Map.cc 
-                      src/SmallBlurryImage.cc src/ShiTomasi.cc src/MapMaker.cc src/MapMakerBase.cc src/MapMakerClientBase.cc src/MapMakerServerBase.cc 
-                      src/PatchFinder.cc src/Relocaliser.cc src/Tracker.cc src/KeyFrameViewer.cc 
-                      src/BundleAdjusterBase.cc src/MiniPatch.cc src/BundleAdjusterMulti.cc src/ChainBundle.cc)
+  CATKIN_DEPENDS roscpp
+                 image_transport
+                 sensor_msgs
+                 cv_bridge geometry_msgs
+                 tf
+                 message_filters
+                 std_msgs
+                 std_srvs
+                 visualization_msgs
+                 dynamic_reconfigure
+                 pcl_ros
+                 message_runtime
+  DEPENDS SuiteSparse
+          CVD
+          OpenGL
+          GVars3
+)
+
+set(MCPTAM_SRC
```

Comment:
I fail to see how this makes things easier, other than putting each file on a separate line ( I do like that).  There are still redundant lists of the same .cc files. Can we organize these in topics that are or aren't used in the executables, like gui, cv, calib, mapping, etc?